### PR TITLE
update go and modules. fix vulnerabilities

### PR DESCRIPTION
**Pull request by stafot**
#### Summary
update go and modules. fix vulnerabilities

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-7444

#### Release Note

```release-note
update go and modules. fix vulnerabilities
```

### CODE REVIEW

File: .github/workflows/.test.yml [hunk 1]
```diff
@@ -8,19 +8,21 @@ jobs:
...
     container:
-      image: golang:1.20-bookworm
+      image: golang:1.21-bookworm
     steps:
     - name: ci/checkout-repo
       uses: actions/checkout@v4
     - name: ci/setup-go-and-cache
       uses: ./.github/actions/go-setup
+    - name: Install goimports
+      run: go install golang.org/x/tools/cmd/goimports@latest
```
Comment (gabrieljackson) [inline .github/workflows/.test.yml:18]: Can you help me understand this part? I see that there is also a Makefile target for this.

Comment (stafot) [reply .github/workflows/.test.yml:18]: For some reason make command didn't work on the github action context so I forced to use this.  These changes in github actions come from a purpose to unify as much as possible the GAs across our repositories to be more maintainable in the long run. As far as I remember this step was required for next steps to work (check-style). Don't remember though why it failed without this.

Comment (stafot) [reply .github/workflows/.test.yml]: I removed `goimports` as go fmt is enough

**Review by gabrieljackson — APPROVED** [review]
LGTM

**Comment by stafot** [comment]
gentle ping @nickmisasi @stylianosrigas

File: Makefile [hunk 2]
```diff
@@ -92,21 +117,21 @@ govet: ## Runs govet against all packages.
...
+#		if [ "$$files" ]; then \
+#			goimports_output=$$(goimports -d $$files 2>&1); \
+#			if [ "$$goimports_output" ]; then \
+#				echo "$$goimports_output"; \
+#				echo "goimports failure"; \
+#				exit 1; \
+#			fi; \
+#		fi; \
+#	done
+#	@echo "goimports success"; \
```
Comment (nickmisasi) [inline Makefile:134]: If it's commented out, can we remove?

**Review by nickmisasi — APPROVED** [review]

**Pull request merged**